### DISCUSSION TITLE
MACRO MODEL: what the unit of EC?

### DISCUSSION THREAD

**Discussion by my0828** [iiasa/message_ix#752] [MESSAGE modeling]
### in the text of MESSAGE-MACRO_run,gms. I  can't understand why the EC is multiplied by 1000 in the line of code below, as far as I know, the unit of EC is G$ and the unit of GDP is T$. I think the EC should be divided by 1000.

`GDP.L(node_macro,year) = (I.L(node_macro,year) + C.L(node_macro,year) + EC.L(node_macro,year)) * 1000 ;`
`

**Comment by glatterf42** [comment]
Thank you for pointing this out! I have just noticed that our documentation might be lacking in that regard. Because you can find the description of the MACRO variables `I`, `C`, and `EC` [here](https://docs.messageix.org/en/latest/model/MACRO/macro_core.html#decision-variables), but this description does not include their units. 
However, this documentation page indicates that it was created automatically from `message_ix/model/MACRO/macro_core.gms` and [there](https://github.com/iiasa/message_ix/blob/248bcbdef3584323848b9ce984e995011882eeec/message_ix/model/MACRO/macro_core.gms#L86-L103) we can find the definition including the units: `I`, `C`, and `EC` are all defined in `Trillion $`. `GDP` does not seem to have a default unit, it just states "in market exchange rates", which I understand as "whichever units suit the scale of the problem". The line that you mention multiplies variables in `Trillion $` with 1000, converting them to `Billion $`, or `G$`, if you prefer. So `GDP` would have the unit `G$` by default.
